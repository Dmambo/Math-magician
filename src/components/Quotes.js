import React from 'react';
import './styles/Quotes.css';

const Quotes = () => (
  <div className="quote">
    <p className="quoteText">
      &rsquo;Mathematics is the language in which God has written the
      universe&rsquo; - Galileo Galilei
    </p>
  </div>
);

export default Quotes;
