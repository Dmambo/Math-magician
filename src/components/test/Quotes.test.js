import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Quotes from '../Quotes';

test('Quotes gets user interaction', () => {
  render(
    <BrowserRouter>
      <Quotes />
    </BrowserRouter>,
  );
  const linkElement = screen.getByText(/Galileo Galilei/i);
  expect(linkElement).toBeInTheDocument();
});
// test Quotes with snapshot

test('Quotes renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Quotes />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
