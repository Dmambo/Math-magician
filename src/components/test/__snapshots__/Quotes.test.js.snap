// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes renders correctly 1`] = `
<div
  className="quote"
>
  <p
    className="quoteText"
  >
    ’Mathematics is the language in which God has written the universe’ - Galileo Galilei
  </p>
</div>
`;
